receiver/prometheusreceiver/internal/metrics_adjuster_test.go

```diff
@@ -0,0 +1,382 @@
+	}, {
+		"Cumulative: round 2 - instance adjusted based on round 1",
+		[]*metricspb.Metric{cumulative(k1k2, timeseries(2, v1v2, double(2, 66)))},
+		[]*metricspb.Metric{cumulative(k1k2, timeseries(1, v1v2, double(2, 22)))},
+	}, {
+		"Cumulative: round 3 - instance reset (value less than previous value), adjusted should be empty",
+		[]*metricspb.Metric{cumulative(k1k2, timeseries(3, v1v2, double(3, 55)))},
+		[]*metricspb.Metric{},
+	}, {
+		"Cumulative: round 4 - instance adjusted based on round 3",
+		[]*metricspb.Metric{cumulative(k1k2, timeseries(4, v1v2, double(4, 72)))},
+		[]*metricspb.Metric{cumulative(k1k2, timeseries(3, v1v2, double(4, 17)))},
+	}}
+	runScript(t, script)
+}
+
+func Test_cumulativeDistribution(t *testing.T) {
+	script := []*metricsAdjusterTest{{
+		"CumulativeDist: round 1 - initial instance, adjusted should be empty",
+		[]*metricspb.Metric{cumulativeDist(k1k2, timeseries(1, v1v2, dist(1, bounds0, []int64{4, 2, 3, 7})))},
+		[]*metricspb.Metric{},
+	}, {
+		"CumulativeDist: round 2 - instance adjusted based on round 1",
+		[]*metricspb.Metric{cumulativeDist(k1k2, timeseries(2, v1v2, dist(2, bounds0, []int64{6, 3, 4, 8})))},
+		[]*metricspb.Metric{cumulativeDist(k1k2, timeseries(1, v1v2, dist(2, bounds0, []int64{2, 1, 1, 1})))},
+	}, {
+		"CumulativeDist: round 3 - instance reset (value less than previous value), adjusted should be empty",
+		[]*metricspb.Metric{cumulativeDist(k1k2, timeseries(3, v1v2, dist(3, bounds0, []int64{5, 3, 2, 7})))},
+		[]*metricspb.Metric{},
+	}, {
+		"CumulativeDist: round 4 - instance adjusted based on round 3",
+		[]*metricspb.Metric{cumulativeDist(k1k2, timeseries(4, v1v2, dist(4, bounds0, []int64{7, 4, 2, 12})))},
+		[]*metricspb.Metric{cumulativeDist(k1k2, timeseries(3, v1v2, dist(4, bounds0, []int64{2, 1, 0, 5})))},
+	}}
+	runScript(t, script)
+}
+
+func Test_summary(t *testing.T) {
+	script := []*metricsAdjusterTest{{
+		"Summary: round 1 - initial instance, adjusted should be empty",
+		[]*metricspb.Metric{summary(k1k2, timeseries(1, v1v2, summ(1, 10, 40, percent0, []float64{1, 5, 8})))},
+		[]*metricspb.Metric{},
+	}, {
+		"Summary: round 2 - instance adjusted based on round 1",
+		[]*metricspb.Metric{summary(k1k2, timeseries(2, v1v2, summ(2, 15, 70, percent0, []float64{7, 44, 9})))},
+		[]*metricspb.Metric{summary(k1k2, timeseries(1, v1v2, summ(2, 5, 30, percent0, []float64{7, 44, 9})))},
+	}, {
+		"Summary: round 3 - instance reset (count less than previous), adjusted should be empty",
+		[]*metricspb.Metric{summary(k1k2, timeseries(3, v1v2, summ(3, 12, 66, percent0, []float64{3, 22, 5})))},
+		[]*metricspb.Metric{},
+	}, {
+		"Summary: round 4 - instance adjusted based on round 3",
+		[]*metricspb.Metric{summary(k1k2, timeseries(4, v1v2, summ(4, 14, 96, percent0, []float64{9, 47, 8})))},
+		[]*metricspb.Metric{summary(k1k2, timeseries(3, v1v2, summ(4, 2, 30, percent0, []float64{9, 47, 8})))},
+	}}
+	runScript(t, script)
+}
+
+func Test_multiMetrics(t *testing.T) {
```

Comment:
MultiMetrics test covers everything that individual metric test covers. may be just keep the mulitmetric test.

Reply:
I considered this as well  - the individual tests are easier for debugging purposes and documenting what should happen for the different aggregation types.

Unless you object, I propose keeping them for now - if maintaining them is problematic, we can remove them subsequently.